### PR TITLE
Adds Teaspoon JavaScript testing framework to project.

### PR DESCRIPTION
**Pull request by mejackreed**
- Adds jQuery gem into project, just so it can be injected into Teaspoon tests
 - Modifies default rake task to run teaspon and rspec
 - Create js specs for image and file, that change for included modules to be defined

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -8,4 +8,4 @@ env:
   global:
     - NOKOGIRI_USE_SYSTEM_LIBRARIES=true
 
-script: bundle exec rspec spec
\ No newline at end of file
+script: bundle exec rake
```
Comment (cbeer) [inline .travis.yml]: This should be the default -- remove it?

**Comment by mejackreed** [comment]
@cbeer thanks for that... should be good to go.

@jkeck what do you think?

**Comment by jkeck** [comment]
:+1: this will definitely make it easier for you guys to start writing useful JS unit tests as you start making enhancements/changes.

**Comment by jkeck** [comment]
@mejackreed needs a rebase.

**Pull request merged**